Graphics/GLSLTools/src/DXILUtils.cpp

```diff
@@ -0,0 +1,378 @@
+/*
+ *  Copyright 2019-2020 Diligent Graphics LLC
+ *  Copyright 2015-2019 Egor Yusov
+ *  
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  In no event and under no legal theory, whether in tort (including negligence), 
+ *  contract, or otherwise, unless required by applicable law (such as deliberate 
+ *  and grossly negligent acts) or agreed to in writing, shall any Contributor be
+ *  liable for any damages, including any direct, indirect, special, incidental, 
+ *  or consequential damages of any character arising as a result of this License or 
+ *  out of the use or inability to use the software (including but not limited to damages 
+ *  for loss of goodwill, work stoppage, computer failure or malfunction, or any and 
+ *  all other commercial damages or losses), even if such Contributor has been advised 
+ *  of the possibility of such damages.
+ */
+
+#include <unordered_set>
+#include <unordered_map>
+#include <memory>
+#include <array>
+#include <locale>
+#include <cwchar>
+
+#ifdef WIN32
+#    include <Unknwn.h>
+#    include <guiddef.h>
+#    include <atlbase.h>
+#    include <atlcom.h>
+#endif
+
+#include "../../ThirdParty/dxc/dxcapi.h"
+
+#include "DXILUtils.hpp"
+#include "DataBlobImpl.hpp"
+#include "RefCntAutoPtr.hpp"
+
+namespace Diligent
+{
+
+// Implemented in GLSLSourceBuilder.cpp
+const char* GetShaderTypeDefines(SHADER_TYPE Type);
+
+namespace
+{
+
+// clang-format off
+static const char g_HLSLDefinitions[] =
+{
+#include "../../GraphicsEngineD3DBase/include/HLSLDefinitions_inc.fxh"
+};
+// clang-format on
+
+struct VkDXILCompilerLib
+{
+    HMODULE               Module         = nullptr;
+    DxcCreateInstanceProc CreateInstance = nullptr;
+    ShaderVersion         MaxShaderModel{6, 5};
+
+    VkDXILCompilerLib()
+    {
+        Module = LoadLibraryA("vk_dxcompiler.dll");
```

Comment:
This code is windows-specific. I am trying not to mix code related to different platforms in one file. VkDXILCompilerLib should contain functionality common to all platforms, and there should be `VkDXILCompilerLibWin32`, `VkDXILCompilerLibLinux`, `VkDXILCompilerLibMacOS` class that implement platform specific functionality (library loading, entry function export, etc.)